### DISCUSSION TITLE
Integrate with AWS MediaPackage

### DISCUSSION THREAD

**Discussion by AffiTheCreator** [lookfirst/sardine#356] [Q&A]
I'm trying to send some m3u8 , ts and webvtt files to a MediaPackage HLS Ingest endpoint, but I keep getting a 404 every time i try to push a file. 

I have setup the sardine client with the username and password provided by the MediaPackage channel, have you tried doing this scenario? If so, how did you overcome this issue?

**Comment by lookfirst** [comment]
Hi there, thanks for the issue. I honestly have no idea.

404 sounds like it is either an authentication or some other error with the URL that you're using.

Are there any headers that are returned as well?

Since this likely isn't an issue with Sardine itself, I'm going to move this to a discussion in the hopes that someone else sees it and can contribute feedback. If there is something specific about Sardine that you find needs fixing, please open another issue for that.

**Comment by AffiTheCreator** [comment]
Hi

So after some testing we concluded that mediapackage needs a md5 autentication in the header and sardine does not support such authentication, hence the not working.
We ended up doing the integration with java.Requests.... aws also requires the files to be renamed to channel_<whatever>.<extension>

So in case anyone is trying to use sardine to send files to mediapackage either use the AWS sdk (works great) or a normal requests api call should work. 
Unfortunately, sardine does not support the authentication required by AWS.

**Comment by lookfirst** [comment]
Thank you for your update.

Sardine supports any sort of modifications to headers and files that you want. It gives you access to the underlying HTTP library and you choose the names of the files.

But yea, probably best to use the AWS SDK as that'll provide easier interfaces and be supported better.
